### PR TITLE
capture: `CaptureResult` can be a namedtuple again

### PR DESCRIPTION
**Pull request by bluetech**
mypy now supports generic NamedTuple.

### CODE REVIEW

File: src/_pytest/capture.py [hunk 1]
```diff
@@ -492,59 +492,19 @@ def writeorg(self, data):
...
-@final
-@functools.total_ordering
-class CaptureResult(Generic[AnyStr]):
-    """The result of :method:`CaptureFixture.readouterr`."""
+# Generic NamedTuple only supported since Python 3.11.
+if sys.version_info >= (3, 11) or TYPE_CHECKING:
 
-    __slots__ = ("out", "err")
+    @final
+    class CaptureResult(NamedTuple, Generic[AnyStr]):
```
Comment (RonnyPfannschmidt) [inline src/_pytest/capture.py:499]: lovely

**Review by RonnyPfannschmidt — APPROVED** [review]

**Comment by bluetech** [comment]
Some issues with the runtime typing on older python versions, I tried something else for it, let's see if this one works.

**Comment by bluetech** [comment]
@RonnyPfannschmidt I needed to make it slightly uglier, but still much better than before. Would appreciate another look.

File: src/_pytest/capture.py [hunk 2]
```diff
@@ -492,59 +492,24 @@ def writeorg(self, data):
...
-        return hash(tuple(self))
-
-    def __lt__(self, other: object) -> bool:
-        if not isinstance(other, (CaptureResult, tuple)):
-            return NotImplemented
-        return tuple(self) < tuple(other)
-
-    def __repr__(self) -> str:
-        return f"CaptureResult(out={self.out!r}, err={self.err!r})"
+        __slots__ = ()
```
Comment (RonnyPfannschmidt) [inline src/_pytest/capture.py:512]: I wonder if attaching type annotations to that class works

**Review by RonnyPfannschmidt — APPROVED** [review]
Gnarly but necessary hack

**Review by nicoddemus — APPROVED** [review]

File: src/_pytest/capture.py (hunk 2 above)
Comment (bluetech) [reply src/_pytest/capture.py:512]: Just checked: nope, doesn't work as expected.

**Pull request merged**